enhancements/installer/openstack-no-fip-install.md

```diff
@@ -0,0 +1,136 @@
+---
+title: openstack-bring-your-own-external-connectivity
+authors:
+  - "@egarcia"
+  - "@adduarte"
+reviewers:
+  - "@mandre"
+  - "@pierre"
+approvers:
+  - "@mandre"
+  - "@pierre"
+creation-date: 2020-5-26
+last-updated: 2020-5-26
+status: implementable
+---
+
+# OpenStack BYO External Connectivity
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Numerous customers have requested the ability to run the OpenShift on OpenStack IPI installer without using Floating IP addresses.
+In the installer, we use floating IPs as a way to set up exeternal connectivity to the OpenShift cluster. Depending on how the OpenStack cluster
+is set up, the use of floating IPs may be impossible or very inconvenient. In order to support these customers, we intend to support installations
+on these types of OpenStack clusters.
+
+## Motivation
+
+### Goals
+
+- Extend the installer to support provider networks
+- Allow the installer to be run without setting up external connectivity
+
+### Non-Goals
+
+This enhancement does not intend to provide any guidance or features that go further than allowing customers to deploy without Floating IPs. Operations like attaching loadbalancers to the entrypoints, or any day 2 operations are beyond the scope of this enhancement. 
+
+## Proposal
+
+In order to implement this feature fully, the following changes must be made:
+1. Make `externalNetwork` an optional argument in the install-config when `machinesSubnet` is set
+2. Disable `lbFloatingIP` when no external network is passed
+3. Make the API and ingress VIPs dynamically assigned by default, and only assign fixed IPs if they are specified in the install config
+
+### Design Details
```

Comment:
Do we have to do anything with DNS?

Reply:
We rely on the user-provided subnet identified by `machinesSubnet` to provide valid DNS resolvers via DHCP. 

https://github.com/openshift/installer/blob/master/docs/user/openstack/customization.md#custom-subnets